### ISSUE TITLE
Indexing into `Word`s and `views`

### ISSUE THREAD

**Issue by mikolajpabiszczak** [kalmarek/KnuthBendix.jl#14]
To consider for later: right now when we iterate over `Word` we obtain Integers. If we subset `Word` via iterator we obtain a vector. Maybe it is sensible to always obtain `Word` via that protocol, i.e. `Word([1,2,3])[2] = Word([2])` and `Word([1,2,3])[2:3] = Word([2,3])`. It would make the code for `kbs1` clearer (shall I open an issue to discuss it separately? This is not something important and maybe later it will become more evident in which way this should be solved).

**Comment by kalmarek** [comment]
my take on this would be to keep it as close to vectors as possible:
* keep `Word([1,2,3])[2]` as an integer i.e. pointer to letter
* `Word([1,2,3])[2:2]` as a word (indexing with a `Range`/`AbstractVector`)
compare:
```julia
julia> v = [1,2,3]
3-element Array{Int64,1}:
 1
 2
 3

julia> v[2]
2

julia> v[2:2]
1-element Array{Int64,1}:
 2

```

we should also think about the `view` syntax into words:
```julia
julia> v = collect(1:7)
7-element Array{Int64,1}:
 1
 2
 3
 4
 5
 6
 7

julia> w = view(v, 2:4)
3-element view(::Array{Int64,1}, 2:4) with eltype Int64:
 2
 3
 4

julia> w[2] = 100
100

julia> w
3-element view(::Array{Int64,1}, 2:4) with eltype Int64:
   2
 100
   4

julia> v
7-element Array{Int64,1}:
   1
   2
 100
   4
   5
   6
   7

```

**Comment by mikolajpabiszczak** [comment]
It seems wise: that way the code in e.g. `knuthbendix1` would be simpler and more self-evident and we would not destroy the way `Ordering`s work at the moment.

**Comment by kalmarek** [comment]
closed by #16
